am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -122,4 +126,41 @@ public FilterResourceResponse filterResource(String userId, String resourceId, J
 
         return null;
     }
+
+
+    /**
+     * Returns a map of JsonPointers and set of permissions when record with user role and type exists in database.
+     *
+     * @param serviceName  (service_name)
+     * @param resourceType (resource_type)
+     * @param resourceName (resource_name)
+     * @param roleNames    A list of (role_names)
+     * @return permissionsByTypeAndRole or null
+     */
+    public Map<JsonPointer, Set<Permission>> getRolePermissions(
+        String serviceName, String resourceType, String resourceName, @NonNull List<String> roleNames) {
+        if (roleNames.size() > ROLENAMESIZE) {
+            throw new UnsupportedOperationException();
```

Comment:
Chosen exception `UnsupportedOperationException` communicates that operation is not supported at this point while here operations is supported as long as only one role is passed in. Please communicate programmer mistake in more clear manner. Choose either different exception or provide reason why operation is not supported.

Reply:
Has not implemented exception at first but PMD does not allow this. Will add in context now.

Reply:
👍

Reply:
👍